devicetypes/smartthings/zigbee-accessory-dimmer.src/zigbee-accessory-dimmer.groovy

```diff
@@ -42,64 +43,140 @@ metadata {
 	}
 }
 
-def getSTEP() {10}
+def getSTEP() { 10 }
+
+def getSCENES_CLUSTER() { 0x0005 }
+def getSCENES_STORE_SCENE_COMMAND() { 0x04 }
+def getSCENES_RECALL_SCENE_COMMAND() { 0x05 }
+
+def getONOFF_ON_COMMAND() { 0x01 }
+def getONOFF_OFF_COMMAND() { 0x00 }
+def getLEVEL_MOVE_LEVEL_COMMAND() { 0x00 }
+def getLEVEL_MOVE_COMMAND() { 0x01 }
+def getLEVEL_STEP_COMMAND() { 0x02 }
+def getLEVEL_STOP_COMMAND() { 0x03 }
+def getLEVEL_MOVE_LEVEL_ONOFF_COMMAND() { 0x04 }
+def getLEVEL_MOVE_ONOFF_COMMAND() { 0x05 }
+def getLEVEL_STEP_ONOFF_COMMAND() { 0x06 }
+def getLEVEL_STOP_ONOFF_COMMAND() { 0x07 }
+def getLEVEL_DIRECTION_UP() { "00" }
+def getLEVEL_DIRECTION_DOWN() { "01" }
+
+def getUINT8_STR() { "20" }
+
+
+def isIkeaDimmer() {
+	device.getDataValue("model") == "TRADFRI wireless dimmer"
+}
+def isAuroraRemote() {
+	device.getDataValue("model") == "Remote50AU"
+}
+def isMullerLichtTint() {
+	device.getDataValue("model") == "ZBT-DIMController-D0800"
+}
 
 // Parse incoming device messages to generate events
 def parse(String description) {
 	log.debug "description is $description"
+	def results = []
 
 	def event = zigbee.getEvent(description)
 	if (event) {
 		if (event.name=="level" && event.value==0) {}
 		else {
-			sendEvent(event)
+			results << createEvent(event)
 		}
 	} else {
 		def descMap = zigbee.parseDescriptionAsMap(description)
-		if (descMap && descMap.clusterInt == 0x0006) {
-			if (descMap.commandInt == 0x01 || descMap.commandInt == 0x00) {
+
+		if (descMap && descMap.clusterInt == zigbee.ONOFF_CLUSTER) {
```

Comment:
You can remove the check of `descMap` since it will always be a map and simplify the if conditionals to just `descMap.clusterInt == zigbee.ONOFF_CLUSTER`